scripts/test-monitor.sh

```
@@ -0,0 +1,325 @@
+    
+    # Run tests and capture output
+    npm test -- --json 2>&1 | tee "$log_file" > /dev/null
+    
+    # Extract summary from Jest JSON output
+    local json_output=$(grep '^{' "$log_file" | tail -1)
+    
+    if [ -n "$json_output" ]; then
+        echo "$json_output" > "$SUMMARY_FILE"
+        
+        # Parse results
+        local total_suites=$(echo "$json_output" | grep -oE '"numTotalTestSuites":[0-9]+' | cut -d: -f2)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)